src/node.rs

```diff
@@ -93,78 +88,95 @@ impl<S: StorageExt> SawtoothRaftNode<S> {
             leader_state: None,
             follower_state: Some(FollowerState::Idle),
             block_queue: BlockQueue::new(),
-            raft_id_to_peer_id: peers.into_iter().map(|peer_id| (peer_id_to_raft_id(&peer_id), peer_id)).collect(),
+            raft_id_to_peer_id: peers
+                .into_iter()
+                .map(|peer_id| (peer_id_to_raft_id(&peer_id), peer_id))
+                .collect(),
             period,
         }
     }
 
     pub fn on_block_new(&mut self, block: Block) {
         if match self.leader_state {
-            Some(LeaderState::Publishing(ref block_id)) => {
-                block_id == &block.block_id
-            },
+            Some(LeaderState::Publishing(ref block_id)) => block_id == &block.block_id,
             _ => false,
         } {
-            debug!("Leader({:?}) transition to Validating block {:?}", self.peer_id, block.block_id);
+            debug!(
+                "Leader({:?}) transition to Validating block {:?}",
+                self.peer_id, block.block_id
+            );
             self.leader_state = Some(LeaderState::Validating(block.block_id.clone()));
         }
 
         debug!("Block has been received: {:?}", &block);
         self.block_queue.block_new(block.clone());
 
-        self.service.check_blocks(vec![block.block_id]).expect("Failed to send check blocks");
+        self.service
+            .check_blocks(vec![block.block_id])
+            .expect("Failed to send check blocks");
     }
 
     pub fn on_block_valid(&mut self, block_id: BlockId) {
         // Handle out of order new/valid updates
         debug!("Block has been validated: {:?}", &block_id);
         self.block_queue.block_valid(&block_id);
         if match self.leader_state {
-            Some(LeaderState::Publishing(ref expected)) => {
-                expected == &block_id
-            },
-            Some(LeaderState::Validating(ref expected)) => {
-                expected == &block_id
-            },
+            Some(LeaderState::Publishing(ref expected)) => expected == &block_id,
+            Some(LeaderState::Validating(ref expected)) => expected == &block_id,
             _ => false,
         } {
-            debug!("Leader({:?}) transition to Proposing block {:?}", self.peer_id, block_id);
+            debug!(
+                "Leader({:?}) transition to Proposing block {:?}",
+                self.peer_id, block_id
+            );
             info!("Leader({:?}) proposed block {:?}", self.peer_id, block_id);
-            self.raw_node.propose(vec![], block_id.clone().into()).expect("Failed to propose block to Raft");
+            self.raw_node
+                .propose(vec![], block_id.clone())
+                .expect("Failed to propose block to Raft");
             self.leader_state = Some(LeaderState::Proposing(block_id));
         }
     }
 
+    #[allow(clippy::ptr_arg)]
```

Comment:
Should this be fixed instead of allowed?

Reply:
Warning is because of BlockId Vec![], and here we have &Vec![] instead of &[...]. Allowing clippy warning here seems to be fine. I am open for suggestions.